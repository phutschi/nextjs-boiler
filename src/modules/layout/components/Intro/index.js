import * as React from 'react';

import theme from './main.scss';

type Props = {
    children : React.Node
}

const Intro = ( { children }: Props ) => (
    <div className={theme.intro}>
        {children}
    </div>
);

export default Intro;
