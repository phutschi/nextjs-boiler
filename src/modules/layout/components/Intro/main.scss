.intro {
    position: relative;
    margin: 30vh auto 0;
    width: 100%;
}
