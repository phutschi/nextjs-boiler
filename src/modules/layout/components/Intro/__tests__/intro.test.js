/* global expect */

import * as React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Intro from '../';

describe( '<Intro />', () => {
    it( 'Snapshot', () => {
        const wrapper = renderer.create( <Intro /> ).toJSON();

        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'Renders with child components', () => {
        const wrapper = shallow(
            <Intro>
                <div className="intro-child" />
            </Intro>,
        );

        expect( wrapper.find( 'div.intro' ) ).toHaveLength( 1 );
        expect( wrapper.find( 'div.intro-child' ) ).toHaveLength( 1 );
    } );
} );
