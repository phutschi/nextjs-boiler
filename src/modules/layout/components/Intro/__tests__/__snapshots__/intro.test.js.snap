// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Intro /> Snapshot 1`] = `
<div
  className="intro"
/>
`;
